src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj

```diff
@@ -88,6 +88,18 @@
       <Output TaskParameter="ItemsWithFullName" ItemName="_Dependency"/>
     </Microsoft.DotNet.Arcade.Sdk.GetAssemblyFullName>
 
+    <!--
+      List of of private dependencies we insert inside our VSIX that need to have a code base version updated.
+      The roslyn insertion tool will read these assembly versions and update AssemblyVersions.tt file in VS
+      using the AssemblyVersionName appended with 'Version'.
+
+      These are separate from the _Dependency list as we do not need to add the package dependency.
+    -->
+    <ItemGroup>
+      <_AdditionalDependencyVersion Include="@(_Dependency)" Condition="$([MSBuild]::ValueOrDefault('%(Identity)', '').StartsWith('Microsoft.VisualStudio.LanguageServer.Protocol'))"
```

Comment:
>$([MSBuild]::ValueOrDefault('%(Identity)', '').StartsWith('Microsoft.VisualStudio.LanguageServer.Protocol')) [](start = 72, length = 108)

You expect this to be exact match, so let's compare %(identity) with the exact name of the dll

Reply:
I wonder if this would work:

```
 <_AssemblyVersionVariable Include="RoslynLanguageServerProtocol" Condition="'%(_Dependency.Identity)' == 'Microsoft.VisualStudio.LanguageServer.Protocol.dll'">
  <Version>$([MSBuild]::ValueOrDefault('%(_Dependency.FusionName)', '').Split(',')[1].Split('=')[1])</Version>
 </_AssemblyVersionVariable>
```

or if not:

```
 <_AssemblyVersionVariableInclude="@(_Dependency->'RoslynLanguageServerProtocol')" Condition="'%(_Dependency.Identity)' == 'Microsoft.VisualStudio.LanguageServer.Protocol.dll'">
  <Version>$([MSBuild]::ValueOrDefault('%(_Dependency.FusionName)', '').Split(',')[1].Split('=')[1])</Version>
 </_AssemblyVersionVariable>
```

then we could also add 

```
 <_AssemblyVersionVariableInclude="Roslyn" Version="$(AssemblyVersion)" />
```

and also the other dependencies, once we filter out the VS ones:

```
 <_AssemblyVersionVariableInclude="@(_Dependency->'%(_AssemblyName)')" Version="%(_Dependency._AssemblyVersion))" />
```

This would make a nice separate item group that would contain all the versions we want to write to csv file.


---
In reply to: [409234328](https://github.com/dotnet/roslyn/pull/43245#discussion_r409234328) [](ancestors = 409234328)